plugins/inputs/ipset/ipset.go

```diff
@@ -0,0 +1,106 @@
+// +build linux
+
+package ipset
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Ipsets is a telegraf plugin to gather packets and bytes counters from ipset
+type Ipset struct {
+	ShowAllSets bool
+	UseSudo     bool
+	lister      setLister
+}
+
+// Description returns a short description of the plugin
+func (ipset *Ipset) Description() string {
+	return "Gather packets and bytes counters from Linux ipsets"
+}
+
+// SampleConfig returns sample configuration options.
+func (ipset *Ipset) SampleConfig() string {
+	return `
+  ## By default, we only show sets which have already matched at least 1 packet.
+  ## set show_all_sets = true to gather them all.
+  show_all_sets = false
+  ## Adjust your sudo settings appropriately if using this option ("sudo ipset save")
+  ## TODO: can we replace this with systemd privileges ? CAP_NET_ADMIN should DTRT
```

Comment:
I guess this answers my question above, if we don't know if this will work don't add it.